### DISCUSSION TITLE
in text-web-ui - deepspeed any special to doo?

### DISCUSSION THREAD

**Discussion by kalle07** [erew123/alltalk_tts#467] [Q&A]
if i disable it
....

Output generated in 2.04 seconds (40.69 tokens/s, 83 tokens, context 605, seed 587445349)
Traceback (most recent call last):
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\queueing.py", line 541, in process_events
    response = await route_utils.call_process_api(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\route_utils.py", line 276, in call_process_api
    output = await app.get_blocks().process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\blocks.py", line 1928, in process_api
    result = await self.call_function(
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\blocks.py", line 1514, in call_function
    prediction = await anyio.to_thread.run_sync(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\_backends\_asyncio.py", line 2144, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\utils.py", line 833, in wrapper
    response = f(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\extensions\alltalk_tts\script.py", line 606, in send_deepspeed_request
    process_lock.release()
RuntimeError: release unlocked lock
Traceback (most recent call last):
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\queueing.py", line 541, in process_events
    response = await route_utils.call_process_api(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\route_utils.py", line 276, in call_process_api
    output = await app.get_blocks().process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\blocks.py", line 1928, in process_api
    result = await self.call_function(
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\blocks.py", line 1514, in call_function
    prediction = await anyio.to_thread.run_sync(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\_backends\_asyncio.py", line 2144, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\installer_files\env\Lib\site-packages\gradio\utils.py", line 833, in wrapper
    response = f(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^
  File "e:\text-generation-webui\extensions\alltalk_tts\script.py", line 606, in send_deepspeed_request
    process_lock.release()
RuntimeError: release unlocked lock
...

and if i enalble it it does nothing
how can i check if its runs?

**Comment by erew123** [comment]
The issue is you have installed it into the TGWUI Python Environment, which isnt currently supported, as stated here https://github.com/erew123/alltalk_tts/wiki/Install-%E2%80%90-Text%E2%80%90generation%E2%80%90webui-Installation#read-before-installing-into-tgwuis-python-environment and here https://github.com/erew123/alltalk_tts/issues/377

And you should be installing it as a Standalone installation and using the TGWUI remote extension (detailed on those links)

**Reply by kalle07** [reply]
i see ... but iv first installed the stand-allone version 
i mean clone the git and start the "atsetup.bat"

and than copied this files 
text-generation-webui/
      ├── extensions/
      │   └── alltalk_tts/           # Create this folder
      │       ├── script.py          # Copied from \alltalk_tts\system\TGWUI Extension
      │       ├── settings.py        # Copied from \alltalk_tts\system\TGWUI Extension


should i wait a month untill this TGWUI is running ?

**Reply by erew123** [reply]
I dont know what you have wrong there. I have tested a fresh install of TGWUI today with the AllTalk TGWUI remote extension and it worked fine. My best guess is you have something wrong with your TGWUI Python environment.

![image](https://github.com/user-attachments/assets/5d902478-6888-4f8b-b3ec-874c4f4aca57)

**Reply by kalle07** [reply]
hmmm what meanst 
TGWUI

only that steps

Windows Instructions

    Open Command Prompt and navigate to the TGWUI extensions directory:

    cd /d C:\path\to\text-generation-webui\extensions

    Clone the AllTalk repository:

    git clone -b alltalkbeta https://github.com/erew123/alltalk_tts

    Start the TGWUI Python environment:

    cd ..
    cmd_windows.bat

    Navigate to the AllTalk directory:

    cd extensions\alltalk_tts

    Run the setup script:

    atsetup.bat

    Follow the on-screen instructions to install necessary requirements. 

and than copy the files into web-ui extensions ?



have some elaxed days off   ;)

**Reply by kalle07** [reply]
oh no ^^
i forget 
cmd_windows.bat in text-webu-ui 

ok all again ...

**Reply by erew123** [reply]
It is not currently possible to clone AllTalk into TGWUI's extensions folder and run setup that way.

You should not attempt that and its also stated with the Standalone Installation that you should NOT install AllTalk in another programs Python environment:

![image](https://github.com/user-attachments/assets/15d556e6-7e3f-4092-8537-6d23b71bcea1)

So install AllTalk as a Standalone application, in its own folder, OUTSIDE of any other applications folder then copy the TGWUI remote extension into TGWUI's folder as per the instructions for the TGWUI remote extension. 

![image](https://github.com/user-attachments/assets/c807ae31-8851-43c2-a92b-9544c0c40501)

**Reply by kalle07** [reply]
thx for the full explanation...

sry realy, i dont know how iv done it, but all run fine in "stand-alone" and "text-web-ui" except that load errors and maybe deepspeed

...on start ...

[AllTalk TTS] _ _ _ _____ _ _ _____ _____ ____
[AllTalk TTS] / \ | | |_ | | | | __ | | / |
[AllTalk TTS] / _ \ | | | | |/ ` | | |/ / | | | | _
[AllTalk TTS] / ___ | | | | | (| | | < | | | | ) |
[AllTalk TTS] // __|| ||_,|||_\ || || |____/
[AllTalk TTS]
[AllTalk TTS] Start-up Mode : Standalone mode
[AllTalk TTS] WAV file deletion : Disabled
[AllTalk TTS] Github updated : 20th December 2024 at 21:42 Branch: alltalkbeta
[AllTalk ENG] Transcoding : ffmpeg found
[AllTalk ENG] DeepSpeed version : 0.14.0+ce78a63
[AllTalk ENG] Python Version : 3.11.11
[AllTalk ENG] PyTorch Version : 2.2.1
[AllTalk ENG] CUDA Version : 12.1
[AllTalk ENG]
[AllTalk ENG] Loading XTTS model xttsv2_2.0.3 on cuda
[AllTalk ENG] Model License : https://coqui.ai/cpml.txt
[AllTalk ENG] Model Loadtime: 11.65 seconds
[AllTalk TTS]
[AllTalk TTS] API Address : 127.0.0.1:7851
[AllTalk TTS] Gradio Light: http://127.0.0.1:7852/
[AllTalk TTS] Gradio Dark : [http://127.0.0.1:7852?__theme=dark](http://127.0.0.1:7852/?__theme=dark)
[AllTalk TTS]
[AllTalk TTS] AllTalk WIKI: https://github.com/erew123/alltalk_tts/wiki
[AllTalk TTS] Errors Help : https://github.com/erew123/alltalk_tts/wiki/Error-Messages-List
[AllTalk TTS]
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.f5tts.f5tts_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.parler.parler_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.piper.piper_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.vits.vits_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.xtts.xtts_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.f5tts.f5tts_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.parler.parler_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.piper.piper_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.vits.vits_settings_page - No module named 'extensions'
[AllTalk TTS] Error Module not found: extensions.alltalk_tts.system.tts_engines.xtts.xtts_settings_page - No module named 'extensions'
[AllTalk TTS] Please use Ctrl+C when exiting AllTalk otherwise a
[AllTalk TTS] subprocess may continue running in the background.
[AllTalk TTS]
[AllTalk TTS] Server Ready
....

so maybe i chang it if i know what iam dooing   ;)

**Reply by erew123** [reply]
It is because when AllTalk, as a **full** standalone installation, is within TGWUI's application folder, TGWUI will try to load AllTalk with the TGWUI Python environment. Very complicated to explain the actual code reasons, needless to say, as specified, you should not install AllTalk into the TGWUI folder at this time, you should **only** install the AllTalk TGWUI Remote Extension.

https://github.com/erew123/alltalk_tts/wiki/Understanding-Python-Environments-Simplified

Until you remove the installation of AllTalk from within the TGWUI extensions folder, you will see the error messages you show above.

**Reply by kalle07** [reply]
its is all like you said since the beginning, but as long all is running its okay...
i can train external RVC and us it and also wave snippets,
maybe deepspeed isnt work, but ^^
